tests/js/client/shell/shell-request.js

```diff
@@ -395,12 +397,33 @@ function RequestSuite () {
   };
 }
 
+function RequestSuite () {
+  let suite = {};
+  deriveTestSuite(BaseRequestSuite(), suite, '');
+  return suite;
+}
+
+function RequestSuiteWithSmallChunks () {
+  let suite = {
+    setUpAll: function() {
+      internal.debugSetFailAt("HttpCommTask<T>::readCallback_in_small_chunks");
+    },
+    
+    tearDownAll: function() {
+      internal.debugClearFailAt();
+    },
+  };
+
+  deriveTestSuite(BaseRequestSuite(), suite, '_SmallChunks');
+  return suite;
+}
 
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief executes the test suite
 ////////////////////////////////////////////////////////////////////////////////
 
 jsunity.run(RequestSuite);
+jsunity.run(RequestSuiteWithSmallChunks);
```

Comment:
I think we can only run this suite if the executable is configured with `ENABLE_FAILURE_TESTS`.
Otherwise the test will try to set failure points, but they are not compiled in.